File: TDD Boot Camp/Part 01/Code/OOPrinciples/src/com/oop/abstraction/Iniyan.java
package com.oop.abstraction;

import java.time.Duration;
import java.time.LocalDate;

public class Iniyan {
	public String firstName;
	public String lastName;
	public LocalDate  dateOfBirth;

	public Iniyan() {
		this.firstName = "Iniyan";
		this.lastName = "Sreenivasan";
		this.dateOfBirth = LocalDate.of(2015, 11, 13);
	}
	
	public int getAge() {
		return calculateAge();
	}
	
	public int calculateAge() {
		long duration =  Duration.between(LocalDate.now(), this.dateOfBirth).getSeconds();
		return (int)(duration/(60*60*24*365));
	}
}

File: TDD Boot Camp/Part 01/Code/C#/OOPrinciples/Abstraction/Iniyan.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Abstraction
{
    public class Iniyan
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }

        public Iniyan()
        {
            FirstName = "Iniyan";
            LastName = "Sreenivasan";
            DateOfBirth = new DateTime(2015, 11, 13);
        }

        public int GetAge()
        {
            return CalculateAge();
        }

        public int CalculateAge()
        {
            double duration = (DateTime.Now - DateOfBirth).TotalSeconds;            
            return (int)(duration / (60 * 60 * 24 * 365));
        }
    }
}
